examples/cooperative_preempt.rs

```diff
@@ -16,42 +16,46 @@ fn main() {
     // explicitly yield control if they are going to do something that may take
     // too long (that is usually a loop!)
     //
-    // There are two ways of yielding control:
+    // There are three ways of yielding control:
     //
-    //  * Local::yield_if_needed(), which will yield if the task has run for too
-    //    long. What "too long" means is an implementation detail, but it will be
-    //    always somehow related to the latency guarantees that the task queues want
-    //    to uphold in their `Latency::Matters` parameter (or
-    //    Latency::NotImportant).
+    //  * glommio::executor().yield_if_needed(), which will yield if the current
+    //    task queue has run for too long. What "too long" means is an
+    //    implementation detail, but it will be always somehow related to the
+    //    latency guarantees that the task queues want to uphold in their
+    //    `Latency::Matters` parameter (or Latency::NotImportant).
     //
-    //  * Local::later(), which will yield immediately (execute the rest of the
-    //    function later).
+    //  * glommio::executor().yield_task_queue_now(), works like yield_if_needed()
+    //    but yields unconditionally.
+    //
+    //  * glommio::executor().yield_now(), which unconditional yield the current
```

Comment:
isn't this the same as the one in futures_lite ? Why go through the executor ?

Reply:
Just to be consistent with the other ones really

Reply:
A reader will automatically ask themselves "but what is the difference between that and futures::yield_now()`? So it's good to mention that it is the same, just under a different interface.

And while we are at it, let's add doc links

Reply:
Alright, I did just that and made lots of corrections to our documentation throughout the crate while at it.  We now have better, more uniform comments and documentation with more links as appropriate.